### PR TITLE
Issue #18593.  Various fixes for Backbone.

### PR DESCRIPTION
**Pull request by Spinarooni**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).

http://backbonejs.org/#Collection-set
http://backbonejs.org/#Collection-slice
http://backbonejs.org/#Sync
http://backbonejs.org/#Collection-sort (Add Option)

### CODE REVIEW

File: types/backbone/index.d.ts [hunk 1]
```diff
@@ -427,6 +446,7 @@ declare namespace Backbone {
 
     // SYNC
     function sync(method: string, model: Model, options?: JQueryAjaxSettings): any;
```
Comment (andy-ms) [inline types/backbone/index.d.ts]: I would just use `model: Model | Collection<Model>`.

**Comment by Spinarooni** [comment]
But wouldn't the parameter name be misleading?  model != Collection?

**Comment by andy-ms** [comment]
I guess, but with an overload you can't pass a variable of type `Model | Collection<Model>` even though that would work at runtime.

**Comment by RyanCavanaugh** [comment]
This PR has been open and unchanged 5 days without signoff or complaint. This will be merged by a maintainer soon if there are no objections.

**Pull request merged**